runtime/cudaq/qis/qvector.h

```diff
@@ -66,6 +67,19 @@ class qvector {
   qvector(const std::initializer_list<complex> &list)
       : qvector(std::vector<complex>{list.begin(), list.end()}) {}
 
+  /// @brief Construct a `qvector` in a specific state.
+  // TODO: determine if we want to support state copying.
+  // On one hand, there could be a use case whereby the user want to keep the
+  // state intact. On the other hand, it's prone to excessive memory usage if
+  // not carefully managed.
+  qvector(const cudaq::state &initState) = delete;
```

Comment:
Are you saying the qvector will necessarily make a copy of `initState`? We can just promote the reference to a pointer here, no? That's not as safe as move/swap operations, but it can be documented that the reference has to remain live.

Reply:
I think in general we could but it'd make the simulator implementation a bit trickier since it now needs to track whether the memory is a borrowed one or something it owns. Plus, if user later call a `get_state`, we then need to do a copy since `get_state` follows the move semantics. 

It's this complication that I think a move semantics loop is tidier...